payroll/models/hr_payslip.py

```diff
@@ -523,29 +523,33 @@ def _get_salary_rules(self):
             sorted_rules |= rule_obj.browse(sorted_rule_ids)
         return sorted_rules
 
-    def _compute_payslip_line(self, rule, localdict, lines_dict):
+    def _compute_rule(self, rule, localdict, lines_dict):
         self.ensure_one()
         # check if there is already a rule computed with that code
         previous_amount = rule.code in localdict and localdict[rule.code] or 0.0
-        # compute the amount of the rule
-        amount, qty, rate, computed_name = rule._compute_rule(localdict)
-        rule_total = amount * qty * rate / 100.0
+        # compute the rule to get some values for the payslip line
+        values = rule._compute_rule(localdict)
+        key = rule.code + "-" + str(localdict["contract"].id)
+        return self._get_line_dict(
+            rule, localdict, lines_dict, key, values, previous_amount
+        )
+
+    def _get_line_dict(self, rule, localdict, lines_dict, key, values, previous_amount):
```

Comment:
In keeping with the theme of descriptive method names I think _get_lines_dict() is a better name for this method.